File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque


def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    def bfs(computer):
        visited[computer] = True
        q = deque()
        q.append(computer)
        while q:
            target = q.popleft()
            visited[target] = True
            for i in range(n):
                if computers[target][i] == 1 and target != i and not visited[i]:
                    q.append(i)
                    
    for c in range(n):
        if not visited[c]:
            bfs(c)
            answer += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    public void bfs(int n, int[][] computers, boolean[] visited, int start) {
        Queue<Integer> q = new LinkedList<>();
        q.offer(start);
        visited[start] = true;
        
        while (!q.isEmpty()) {
            int now = q.poll();
            
            for (int i = 0; i < n; i++) {
                if (!visited[i] && computers[now][i] == 1) {
                    q.offer(i);
                    visited[i] = true;
                }
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        boolean[] visited = new boolean[n];
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                bfs(n, computers, visited, i);
                answer++;
            }
        }
        return answer;
    }
}